### DISCUSSION TITLE
Does eclair support transfers in invoice without payment_secret?

### DISCUSSION THREAD

**Discussion by nueno4** [ACINQ/eclair#2509] [Q&A] [answered]
I wanted to do a non-MPP transfer, so I modified the LND to create an invoice with only feature 9(`var_onion_optin`).
I executed the transfer command from eclair with the following connection(testnet):

```
[A:eclair]---->[B:LND]---->[C:LND]

A: sender
B: relayer
C: payee (create a non-MPP invoice)
```

I read the logs for A(sender) and B(relayer). A did not send `update_add_htlc` and B did not receive it.
If I enable `payment_secret` on the invoice(feature 14), the transfer was successful.
Does eclair support transfers in invoice without payment_secret?

When Phoenix(mainnet) is the sender, `update_add_htlc` is sent, but the LND(A) was in error and returned `update_fail_htlc`.
The cause is unknown.

```
[Phoenix]---->[ACINQ]---->???---->[LND(A)]--->[LND(B)]
```

For now, I am trying to make feature 14 (payment_secret) mandatory for eclair/phoenix.

**Comment by t-bast** [comment, marked as answer]
Bolt 11 invoices MUST have a `payment_secret`, even when MPP isn't used, otherwise the payment is rejected. The `payment_secret` is independent of MPP, you should create invoices with features 9 (`var_onion_optin`) and 14 (`payment_secret`), even if the invoice doesn't support 16 (`basic_mpp`).

**Reply by nueno4** [reply]
This was the only description I could find for `payment_secret` in [Bolt11](https://github.com/lightning/bolts/blob/9af622690ebd41b487e4b8ad35578dc8d7521194/11-payment-encoding.md#requirements-3).

```
A reader:
  ...
  - if there is a valid `s` field:
    - MUST use that as [`payment_secret`](04-onion-routing.md#tlv_payload-payload-format)
```

**Reply by nueno4** [reply]
Sorry, I missed this one:

```
A writer:
  ...
  - MUST include exactly one `s` field.
```
